src/UnityExtension/Assets/Editor/GitHub.Unity/UI/TreeControl.cs

```diff
@@ -0,0 +1,465 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Profiling;
+
+namespace GitHub.Unity
+{
+    [Serializable]
+    public class Tree
+    {
+        [SerializeField] public float ItemHeight = EditorGUIUtility.singleLineHeight;
+        [SerializeField] public float ItemSpacing = EditorGUIUtility.standardVerticalSpacing;
+        [SerializeField] public float Indentation = 12f;
+        [SerializeField] public Rect Margin = new Rect();
+        [SerializeField] public Rect Padding = new Rect();
+
+        [SerializeField] private SerializableTexture2D activeNodeIcon = new SerializableTexture2D();
+        public Texture2D ActiveNodeIcon {  get { return activeNodeIcon.Texture; } set { activeNodeIcon.Texture = value; } }
+
+        [SerializeField] private SerializableTexture2D nodeIcon = new SerializableTexture2D();
+        public Texture2D NodeIcon {  get { return nodeIcon.Texture; } set { nodeIcon.Texture = value; } }
+
+        [SerializeField] private SerializableTexture2D folderIcon = new SerializableTexture2D();
+        public Texture2D FolderIcon {  get { return folderIcon.Texture; } set { folderIcon.Texture = value; } }
+
+        [SerializeField] private SerializableTexture2D rootFolderIcon = new SerializableTexture2D();
+        public Texture2D RootFolderIcon {  get { return rootFolderIcon.Texture; } set { rootFolderIcon.Texture = value; } }
+
+        [SerializeField] public GUIStyle FolderStyle;
+        [SerializeField] public GUIStyle TreeNodeStyle;
+        [SerializeField] public GUIStyle ActiveTreeNodeStyle;
```

Comment:
Normally I wouldn't serialize GUIStyles/Textures, instead I load them on demand (typically in OnEnable). This makes it more difficult to iterate on the source styles/textures without recreating the serialized object completely. Nothing particularly wrong with this approach though.